packages/core/src/tracing/instrumentation/protocols/httpClient.js

```diff
@@ -178,20 +178,12 @@ function instrument(coreModule, forceHttps) {
     const skipTracingResult = cls.skipExitTracing({
       isActive,
       extendedResponse: true,
-      skipParentSpanCheck: true,
```

Comment:
Removing skipParentSpanCheck & skipIsTracing and added a new additional flag `checkReducedSpan` and moved all the logic including reduced span and skipParentSpanCheck to cls.skipExitTracing function;